server/src/main/kotlin/fi/oph/ludos/koodisto/KoodistoService.kt

```diff
@@ -0,0 +1,80 @@
+package fi.oph.ludos.koodisto
+
+import fi.oph.ludos.cache.CacheName
+import fi.oph.ludos.exception.LocalizationException
+import org.slf4j.LoggerFactory
+import org.springframework.cache.CacheManager
+import org.springframework.stereotype.Service
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+
+@Service
+class KoodistoService(val koodistoRepository: KoodistoRepository, val cacheManager: CacheManager) {
+    private final val logger: org.slf4j.Logger = LoggerFactory.getLogger(javaClass)
+
+    init {
+        // Schedule cache update every 2 minutes
+        val scheduler = Executors.newScheduledThreadPool(1)
+        scheduler.scheduleAtFixedRate({ updateCache() }, 0, 10, TimeUnit.MINUTES)
```

Comment:
Tässäki vois toi delay olla 10, ni ei tuu alkuun tuplapyyntöö.

```suggestion
        scheduler.scheduleAtFixedRate({ updateCache() }, 10, 10, TimeUnit.MINUTES)
```